demo/admin/src/products/ManufacturerSelectFields.tsx

```diff
@@ -0,0 +1,125 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { Field, FinalFormSelect, SelectField, useAsyncOptionsProps, useFormApiRef } from "@comet/admin";
+import { MenuItem } from "@mui/material";
+import {
+    GQLManufacturerSelectFragment,
+    GQLManufacturersOptionsFragment,
+    GQLProductManufacturersQuery,
+    GQLProductManufacturersQueryVariables,
+} from "@src/products/ManufacturerSelectFields.generated";
+import React from "react";
+import { FieldRenderProps, Form, FormSpy } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+export const manufacturerSelectFragment = gql`
+    fragment ManufacturerSelect on Product {
+        manufacturer {
+            id
+            updatedAt
+            address {
+                country
+            }
+        }
+    }
+`;
+export const manufacturersOptionsFragment = gql`
+    fragment ManufacturersOptions on Manufacturer {
+        id
+        updatedAt
+    }
+`;
+export const productManufacturersQuery = gql`
+    query ProductManufacturers($filter: ManufacturerFilter) {
+        manufacturers(filter: $filter) {
+            nodes {
+                ...ManufacturersOptions
+                id
+                updatedAt
+                address {
+                    country
+                }
+            }
+        }
+    }
+    ${manufacturersOptionsFragment}
+`;
+
+type FormProps = { data?: GQLManufacturerSelectFragment } & FieldRenderProps<string>;
```

Comment:
why this extra data? FieldRenderProps.input.value already contains the value

type FormProps = `FieldRenderProps<GQLManufacturerSelectFragment>;`